contracts/treasury/src/contract.rs

```diff
@@ -86,48 +84,53 @@ fn rescue_tokens(
     }
 
     let mut msgs: Vec<CosmosMsg> = vec![];
-    match token_info {
-        TokenInfo::Native(denom) => {
-            let mut coin = deps
-                .querier
-                .query_balance(env.contract.address, &denom)
-                .unwrap_or(Coin {
-                    denom,
-                    amount: Uint128::zero(),
-                });
-            if !coin.amount.is_zero() {
-                if coin.amount > amount_to_rescue {
-                    coin.amount = amount_to_rescue;
-                }
-                msgs.push(CosmosMsg::Bank(BankMsg::Send {
-                    to_address: to.to_string(),
-                    amount: vec![coin],
-                }));
-            };
-        }
-        TokenInfo::Cw20(token_addr) => {
-            let token_balance: BalanceResponse = deps.querier.query_wasm_smart(
-                token_addr.clone(),
-                &cw20::Cw20QueryMsg::Balance {
-                    address: env.contract.address.to_string(),
-                },
-            )?;
-            if !token_balance.balance.is_zero() {
-                let amount = if token_balance.balance > amount_to_rescue {
-                    amount_to_rescue
-                } else {
-                    token_balance.balance
-                };
-                msgs.push(CosmosMsg::Wasm(WasmMsg::Execute {
-                    contract_addr: token_addr,
-                    msg: to_binary(&Cw20ExecuteMsg::Transfer {
-                        recipient: to.to_string(),
-                        amount,
-                    })
-                    .unwrap(),
-                    funds: vec![],
-                }));
+    // Validate the `token_address`
+    // Here, there are 2 options for `token_address` - native token denom or CW20 token address.
+    // If `token_address` is the native token denom, its length is no longer than 10 chars.
+    // If `token_address` is the CW20 token address, its length is 63 ~ 65 chars.
+    // The reason is that the cosmwasm contract address is the beche32 address - `chain-prefix`(5 ~ 7) + `address`(58 bytes)
+    if token_address.len() < 58 {
```

Comment:
Where does 58 come from? It seems a bit arbitrary. Is there no function that tells us for sure if an address is a contract?